kubernetes/FAQs.md

```diff
@@ -0,0 +1,84 @@
+- [DNS Specifications for Kubernetes](https://github.com/kubernetes/dns/blob/master/docs/specification.md)
+- [CoreDNS Kubernetes plugin documentation](https://coredns.io/plugins/kubernetes/)
+
+## Pre-requisites before troubleshooting
+
+It is important to check whether the Kubernetes Network is stable before starting to troubleshoot DNS. It may appear like DNS isn't functioning when there is a network issue in the Kubernetes cluster. 
+In other words, an issue like "DNS does not work properly" has often its root cause as "Network is not working properly".
+
+Hence, if you're confident that the network is stable and DNS still doesn't work, this FAQ document is applicable. 
+
+## Known Issues
+
+### Some Prometheus data is missing after reload
+
+In CoreDNS v1.2.2 that is provided with K8s v1.12, after editing the ConfigMap,
+CoreDNS will automatically "reload" its configuration file.
+
+he reload causes some metrics on the Prometheus interface to be missing.
+
+## Frequently Asked Questions
+
+#### FAQ-1 : Why am I unable to get an answer to the DNS queries when I'm using ```busybox``` to test?
+
+Busybox has a [known bug](https://github.com/docker-library/busybox/issues/48) and therefore does not work while using for testing DNS queries in Kubernetes.
+The workaround for this is to ensure that the image of Busybox is lower or equal to 1.28.4
+See: [dns can't resolve kubernetes.default and/or cluster.local](https://github.com/kubernetes/kubernetes/issues/66924#issuecomment-411804435) for more details.
+
+#### FAQ-2 : Why isn't the `autopath` plugin in the default configuration?
+
+Autopath requires  that the option `pods verified` is set in Kubernetes's plugin configuration. The `pods verified` takes up significantly more memory. Please look into the [documentation](https://github.com/coredns/coredns/tree/master/plugin/kubernetes#autopath) for more details.
+
+#### FAQ-3 : How do I test the DNS service?
+
+There are two definite ways to test this: 
+
+- Run a pod with Busybox, as described in [Debugging DNS Resolution](https://kubernetes.io/docs/tasks/administer-cluster/dns-debugging-resolution/)
+- Run a DNS tools dedication pod using the following command:
+
+```
+kubectl run -it --rm --restart=Never --image=infoblox/dnstools:latest dnstools
+```
+
+After the pod is started and you get the console propmt, you can run one of those commands:
+
+```
+# nslookup kubernetes.default
+```
+or
+```
+# dig kubernetes.default A
+```
+
+#### FAQ-4 : How do I rollback to kube-dns? I am using kubeadm tool to install the Kubernetes cluster.
+
+The following command will reapply your cluster with existing settings but will replace CoreDNS with kube-dns.
+
+```
+kubeadm upgrade apply <current-installed-version> --feature-gates CoreDNS=false --force
+```
+
+Warning, although this command is correct, it may not work because of ongoing issues in kubeadm.
+
+#### FAQ-5 : Why is my CoreDNS pod in `CrashLoopBackoff` state?
```

Comment:
@chrisohaver I have added a question on the `crashloopbackoff`. Is there anything to be improved upon?

Reply:
kubeadm has a section on this too... i think it has a couple of other possible reasons.  selinux was one was one i think.  We should copy that here.